doc/reference/modules/transactions.xml

```diff
@@ -569,14 +573,100 @@ finally
+                    itself if connection string <literal>Enlist</literal> setting is not
+                    <literal>false</literal>.
+              </para>
+            </listitem>
+            <listitem>
+                <para>
+                    As of NHibernate v5.0, <literal>FlushMode.Commit</literal> requires the configuration setting
+                    <literal>transaction.use_connection_on_system_events</literal> to be true for flushing
+                    from transaction scope commit. Otherwise, it will be your responsibility to flush the session
+                    before completing the scope.
+                </para>
+                <para>
+                    Using <literal>transaction.use_connection_on_system_events</literal> can cause undesired
+                    transaction promotions to distributed: it requires using a dedicated connection for flushing,
+                    and it delays session disposal (if done inside the scope) to the scope disposal. If you want
+                    to avoid this, set this setting to <literal>false</literal> and manually flush your sessions.
+                </para>
+            </listitem>
+            <listitem>
+                <para>
+                    As of NHibernate v5.0, <literal>ConnectionReleaseMode.AfterTransaction</literal> has no more
+                    by default an "immediate" effect with transaction scopes. Previously, it was releasing the
+                    connection from transaction completion events. But this is not officially supported by
+                    Microsoft and this can cause issues especially with distributed transactions.
+                </para>
+                <para>
+                    Since v5.0, by default, the connection will be actually released after the scope disposal at the
+                    first session usage involving a connection, or at the session closing, whichever come first.
+                    Alternatively, you may <literel>Disconnect()</literel> the session. (Requires
+                    <literal>Reconnect()</literal> before re_using the session.)
+                </para>
+                <para>
+                    When using <literal>transaction.use_connection_on_system_events</literal>, if the session is
+                    disposed within the scope, the connection releasing will still occurs from transaction
+                    completion event.
+                </para>
+            </listitem>
+            <listitem>
+                <para>
+                    As of NHibernate v5.0, using transaction scope and trying to use the session connection within
+                    <literal>AfterTransactionCompletion</literal> is forbidden and will raise an exception.
+                    If the setting <literal>transaction.use_connection_on_system_events</literal>
+                    is <literal>false</literal>, it will forbid any connection usage from
```

Comment:
There is no chance to use Envers when transaction scope is used and `transaction.use_connection_on_system_prepare` is set to `false`, right?

If I understood well, Envers uses `BeforeTransactionCompletion` to write audit data to database, but it is forbidden in this case.

Reply:
I do not know Envers details. But if it does also write its audit data on `Flush`, using explicit flushes should allow to work with `transaction.use_connection_on_system_prepare` set to `false`. Otherwise, you should ask or contribute a change in Envers for having it flush its audit data on session flushes, even within a transaction.

Reply:
@fredericDelaporte @IwanowM 
I'm not sure if I follow you but Envers is using TransactionCompletionProcess (and IAfterTransactionCompletionProcess), yes.
https://github.com/nhibernate/nhibernate-envers/blob/4523dfd7521b294d9cd282e8dbdcba04b560c97d/Src/NHibernate.Envers/Synchronization/AuditProcessManager.cs#L24

Reply:
@RogerKratz 
I have created [issue](https://github.com/nhibernate/nhibernate-envers/issues/7) about that. 
Would you have a look at that?